fw/http.c

```diff
@@ -1615,54 +1615,62 @@ do {									\
 	}
 }
 
-static bool
-tfw_http_hm_suspend(TfwHttpResp *resp, TfwServer *srv)
+/**
+ * Try to mark server as suspended.
+ * In case of HM is active do it, otherwise left unchanged.
+ */
+static void
+tfw_http_hm_try_suspend(TfwHttpResp *resp, TfwServer *srv)
 {
 	unsigned long old_flags, flags = READ_ONCE(srv->flags);
-	/*
-	 * We need to count a total response statistics in
-	 * tfw_apm_hm_srv_limit(), even if health monitor is disabled.
-	 * In this case limit calculation won't be accounted.
-	 */
-	bool lim_exceeded = tfw_apm_hm_srv_limit(resp->status, srv->apmref);
 
-	if (!(flags & TFW_SRV_F_HMONITOR))
-		return true;
-	if (!lim_exceeded)
-		return false;
+	while (flags & TFW_SRV_F_HMONITOR) {
 
-	do {
 		old_flags = cmpxchg(&srv->flags, flags,
-				    flags | TFW_SRV_F_SUSPEND);
+				flags | TFW_SRV_F_SUSPEND);
+
 		if (likely(old_flags == flags)) {
 			T_WARN_ADDR_STATUS("server has been suspended: limit "
 					   "for bad responses is exceeded",
 					   &srv->addr, TFW_WITH_PORT,
 					   resp->status);
 			break;
 		}
-		flags = old_flags;
-	} while (flags & TFW_SRV_F_HMONITOR);
 
-	return true;
+		flags = old_flags;
+	}
 }
 
+/**
+* The main function of Health Monioring.
+* Getting response from the server, it updates responses statistics,
+* checks HM limits and makes solution about server health.
```

Comment:
```suggestion
* checks HM limits and makes a decision about server health.
```
BTW ChatGPT amazingly proofreads text and explains the fixes, so you can write better text and learn why

Reply:
I prefer Reverso